### PR TITLE
Python3 fix journalctl collection failed

### PR DESCRIPTION
**Pull request by chunfuwen**

### CODE REVIEW

**Comment by chunfuwen** [comment]
@ldoktor 
@clebergnu 
please help take a look.

**Comment by chunfuwen** [comment]
paste failed log
```
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
Profiler disabled
Journalctl collection failed: the JSON object must be str, not 'bytes'
```

File: avocado/core/sysinfo.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def _get_cursor(self):
         try:
             cmd = 'journalctl --quiet --lines 1 --output json'
             result = process.system_output(cmd, verbose=False)
-            last_record = json.loads(result)
+            last_record = json.loads(astring.to_text(result))
```
Comment (ldoktor) [inline avocado/core/sysinfo.py]: `result.stdout_text` is preferred

File: avocado/core/sysinfo.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def run(self, logdir):
                 cmd = 'journalctl --quiet --after-cursor %s' % self.cursor
                 log_diff = process.system_output(cmd, verbose=False)
                 dstpath = os.path.join(logdir, self.logf)
-                with gzip.GzipFile(dstpath, "w") as out_journalctl:
+                with gzip.GzipFile(dstpath, "wb") as out_journalctl:
```
Comment (ldoktor) [inline avocado/core/sysinfo.py]: yep, log_diff is bytes so binary mode is the way to go.

File: avocado/core/sysinfo.py (hunk 1 above)
Comment (chunfuwen) [reply avocado/core/sysinfo.py]: The prequiste for result.stdout_text is that result should be cmdResult, but here result is just output,which is either bytes or string

**Comment by chunfuwen** [comment]
@ldoktor 
@clebergnu 
Code had been updated accordingly ,please help take a review.

**Pull request merged**